nixos/modules/system/boot/plymouth.nix

```diff
@@ -65,6 +73,14 @@ in
         '';
       };
 
+      deviceOrientation = mkOption {
+        default = "NoRotation";
+        type = types.str;
```

Comment:
In case you keep the typed version for later submission as nixpkgs PR, nixos already has the following type which basically do the same as your assertion: `type = types.enum ["value1" "value1" ]`;`